BUILD.md

```diff
@@ -0,0 +1,212 @@
+
+![Hyperledger Sawtooth](images/sawtooth_logo_light_blue-small.png)
+
+Hyperledger Sawtooth Core Developers Setup Guide
+=============
+If you are planning to contribute code to please review the contributing guide: [CONTRIBUTING.md]
+
+Supported Operating Systems: Ubuntu 16.04 and MacOS
+
+If you are planning to develop or contribute from a computer running a windows
+operating system, we recommend that you install Ubuntu 16.04 in a virtual
+machine manager, such as hyperv or virtualbox, and develop from the guest os.
+
+Step One: Install Requirements
+-------------
+The Sawtooth Core Requirements are:
+- Docker Community Edition ( 17.05.0-ce or newer)
+- Docker compose (version 1.13.0 or newer)
+- Bash (Linux or OSX)
+
+macOS
+=============
+
+Install the latest version of Docker Engine for macOS:
+https://docs.docker.com/docker-for-mac/install/
+
+On macOS Docker Compose is installed automatically when you
+install Docker Engine.
+
+Linux
+=============
+
+Install the latest version of Docker Engine for Linux:
+
+- Install Docker Engine: https://docs.docker.com/engine/installation/linux/ubuntu
+- Install Docker Compose: https://docs.docker.com/compose/install/
+
+**Note:** that the minimum version of Docker Engine necessary is 17.03.0-ce.
+  Linux distributions often ship with older versions of Docker.
+
+**Note:** Add you user to the group `docker` to avoid having to run every
+docker command as a `sudo`. If you do not do this you will need to prefix all
+the commands in section 3,4, and 5 with `sudo`. To add your account to the
+docker group as follows, replace:
+```bash
+$ sudo adduser $USER docker
+```
+
+Configure Proxy (Optional)
+-------------
+
+If you are behind a network proxy, follow these steps before continuing:
+
+Set the following environment variables:
+  - http_proxy
+  - https_proxy
+  - no_proxy
+
+If you are using the Bash shell, run the following commands:
+
+**Warning:** The example URLs and port numbers used below are examples only.
+Please substitute the actual URL, with actual port numbers, used in your
+environment. Contact your network administrator for the information if
+necessary.
+
+```bash
+  $ export http_proxy=http://example-proxy-server.com:3128
```

Comment:
If you use an address like `proxy-server.example`, it is a valid url, but can never be actually installed as a top-level domain (see https://en.wikipedia.org/wiki/.example)